ibis/backends/decompile/compiler.py

```diff
@@ -0,0 +1,145 @@
+import functools
+import io
+import itertools
+
+import black
+
+import ibis.common.graph as graph
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+
+
+@functools.singledispatch
+def translate(op, *args, **kwargs):
+    raise NotImplementedError(op)
+
+
+@translate.register(dt.DataType)
+def datatype(dtype, *args, **kwargs):
+    return str(dtype)
+
+
+@translate.register(sch.Schema)
+def schema(schema, *, names, types):
+    return repr(list(zip(names, map(str, types))))
```

Comment:
Can we use the `dict` literal representation here? The list of tuples repr is super noisy vs the dict.

Reply:
Updated.